src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -4190,143 +4163,146 @@ int SketchObject::addSymmetric(const std::vector<int>& geoIdList, int refGeoId,
-                                    && (refPosId == Sketcher::PointPos::none)) {
-                                    constNew->setValue(-constr->getValue());
-                                }
-
-                                newconstrVals.push_back(constNew);
-                            }
-                        }
-                        else {// three GeoIds intervene in constraint
-                            auto tit = geoIdMap.find(constr->Third);
-
-                            if (tit != geoIdMap.end()) {// Third is also in the list
-                                Constraint* constNew = constr->copy();
-                                constNew->Name = "";
-                                constNew->First = fit->second;
-                                constNew->Second = sit->second;
-                                constNew->Third = tit->second;
-                                if (isStartEndInverted[constr->First]) {
-                                    if (constr->FirstPos == Sketcher::PointPos::start)
-                                        constNew->FirstPos = Sketcher::PointPos::end;
-                                    else if (constr->FirstPos == Sketcher::PointPos::end)
-                                        constNew->FirstPos = Sketcher::PointPos::start;
-                                }
-                                if (isStartEndInverted[constr->Second]) {
-                                    if (constr->SecondPos == Sketcher::PointPos::start)
-                                        constNew->SecondPos = Sketcher::PointPos::end;
-                                    else if (constr->SecondPos == Sketcher::PointPos::end)
-                                        constNew->SecondPos = Sketcher::PointPos::start;
-                                }
-                                if (isStartEndInverted[constr->Third]) {
-                                    if (constr->ThirdPos == Sketcher::PointPos::start)
-                                        constNew->ThirdPos = Sketcher::PointPos::end;
-                                    else if (constr->ThirdPos == Sketcher::PointPos::end)
-                                        constNew->ThirdPos = Sketcher::PointPos::start;
-                                }
-                                newconstrVals.push_back(constNew);
-                            }
-                        }
-                    }
+            };
+
+            if (constr->Third == GeoEnum::GeoUndef) {
+                if (!(constr->Type == Sketcher::Coincident
+                      || constr->Type == Sketcher::Perpendicular
+                      || constr->Type == Sketcher::Parallel || constr->Type == Sketcher::Tangent
+                      || constr->Type == Sketcher::Distance || constr->Type == Sketcher::Equal
+                      || constr->Type == Sketcher::Angle || constr->Type == Sketcher::PointOnObject
+                      || constr->Type == Sketcher::InternalAlignment)) {

```

Comment:
Can we put each condition in separate line? I know that this is how clang-format does (damn you clang-format!) but it is really hard to read. One trick you can use is to put empty comment at the end of each line to force line break.